akamai/util.go

```diff
@@ -97,3 +103,45 @@ func makeCache() *cache.Cache {
 	profilecache := cache.New(5*time.Minute, 10*time.Minute)
 	return profilecache
 }
+
+func setCorrelationID(correlationid string) {
+	//edge.SetupLogging()
+	//log.Printf("[DEBUG] setCorrelationID %s\n", correlationid)
+	edge.SetLogCorrelationId(correlationid)
+
+}
+
+func getCorrelationID() (result *string) {
+	//log.Printf("[DEBUG] getCorrelationID %s\n", LocalLogCorrelationID)
+	//LocalLogCorrelationID = edge.LogCorrelationID
+	result = edge.LogCorrelationID
+	return result
+}
+
+func PrintLogHeader() {
+	//log.Printf("[DEBUG] START PrintLogHeader %s\n", LocalLogCorrelationID)
+	if edge.LogCorrelationID != nil {
+		strLogCorrelationIDValue := *edge.LogCorrelationID
+		log.Printf("[DEBUG] START CORRELATION ID %s\n", strLogCorrelationIDValue)
+		//edge.LogCorrelationID = nil
+		KeepLogCorrelationID = edge.LogCorrelationID
+	}
+}
+
+func PrintLogFooter() {
+	edge.LogCorrelationID = KeepLogCorrelationID
+	if edge.LogCorrelationID != nil {
+		strLogCorrelationIDValue := *edge.LogCorrelationID
+		log.Printf("[DEBUG] END CORRELATION ID %s\n", strLogCorrelationIDValue)
+		edge.LogCorrelationID = nil
+	}
+}
+
+func CreateNonce() string {
```

Comment:
I feel the nonce generation should be in the open-golang library as a common function so that it can be used in all levels of code but there may be reasons that is not advisable.